### PR TITLE
4.3

### PR DESCRIPTION
**Pull request by gowrav-vishwakarma**
Nginx Does not throw dynamic subdomain, $app->url()->absolute()->getBaseURL() will show main server_name set in nginx configuration only. HTTP_HOST is better option.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
-    "name": "atk4/atk4",
+    "name": "xepan/atk4",
```
Comment (romaninsh) [inline composer.json:2]: this change is probably a mistake.

File: lib/Auth/Basic.php [hunk 2]
```diff
@@ -349,7 +349,7 @@ function verifyCredentials($user, $password)
         // Attempt to load user data by username. If not found, return
         // false
 
-        $data = $this->model->newInstance()->tryLoadBy($this->login_field, $user);
+        $data = $this->model->tryLoadBy($this->login_field, $user);
```
Comment (romaninsh) [inline lib/Auth/Basic.php:352]: riskier than original, why exactly the change?

File: lib/Controller/PageManager.php [hunk 3]
```diff
@@ -132,7 +132,7 @@ function parseRequestedURL(){
             if($https)$url.='s';
 
             // 2. Continue building. We are adding hostname next and port.
-            $url .= "://".$_SERVER["SERVER_NAME"];
+            $url .= "://".$_SERVER["HTTP_HOST"];
```
Comment (romaninsh) [inline lib/Controller/PageManager.php:135]: will this still work with apache?

File: lib/Form/Basic.php [hunk 4]
```diff
@@ -214,9 +214,14 @@ function addField($type, $options = null, $caption = null, $attr = null)
             $template=$this->template->cloneRegion('form_line');
             $field = $this->add($class, $options, null, $template);
         } else {
-            if ($insert_into->template->hasTag($name)) {
-                $template=$this->template->cloneRegion('field_input');
-                $options['show_input_only']=true;
+            if ($insert_into->template->hasTag($name)) {                
+                if($this->app->getConfig('form/layout/inputonly',true)){
```
Comment (romaninsh) [inline lib/Form/Basic.php:218]: seems like some hard-code thing.

File: lib/PathFinder.php [hunk 5]
```diff
@@ -205,13 +205,13 @@ public function addDefaultLocations()
...
-                'template'=>$templates_folder,
-            ))->setBasePath(dirname($base_directory).'/shared');
-        }
+        // if (is_dir(dirname($base_directory).'/shared')) {
+        //     $this->shared_location=$this->addLocation(array(
+        //         'php'=>'lib',
+        //         'addons'=>'addons',
+        //         'template'=>$templates_folder,
+        //     ))->setBasePath(dirname($base_directory).'/shared');
+        // }
```
Comment (romaninsh) [inline lib/PathFinder.php:215]: also, probably to big of a change for legacy projects.

File: composer.json (hunk 1 above)
Comment (gowrav-vishwakarma) [reply composer.json:2]: yep ... :) sure it is ... there are a lot of things i didn't wanted to push

File: lib/Controller/PageManager.php (hunk 3 above)
Comment (gowrav-vishwakarma) [reply lib/Controller/PageManager.php:135]: yep

Comment (gowrav-vishwakarma) [reply lib/Controller/PageManager.php:135]: This was only change i wanted to push btw 😆

**Pull request closed without merging**